### ISSUE TITLE
Get task object right after enqueuing

### ISSUE THREAD

**Issue by divan** [fluttercommunity/flutter_downloader#67]
Hi, thanks for the fantastic package.

I'm trying to get DownloadTask object for the newly enqueued download, but can't make it work. My code looks like this:

```dart
    String taskID = await FlutterDownloader.enqueue(
      url: url,
      savedDir: dir,
      showNotification: false,
      openFileFromNotification: false,
    );
    FlutterDownloader.loadTasks().then((tasks) {
      print('[DB] Found ${tasks.length} tasks');
    });
```

Somehow I always get `Found 0 tasks`. I checked the code, and it looks like when `enqueue` returns, object should be in DB, so `loadTasks()` should find it.
I also see that the whole system is working, files are being downloaded (I can see it in iOS simulator Documents folder).

Any idea what I might be doing wrong? Maybe there is another way to get a DownloadTask  object for the newly enqueued task?

Thanks in advance.

**Comment by hnvn** [comment]
Have you tried the example project yet?

**Comment by divan** [comment]
@hnvn yes. It doesn't do the same.

I'm trying to understand why I get this `Found 0 tasks` result. My understanding is that I should get non-empty result from loadTasks(), is that correct?

**Comment by hnvn** [comment]
In iOS implementation, a new task is inserted into DB in a separate thread, it seems that if you call `loadTasks` right after `enqueue` return, that task hasn't been inserted to DB yet.

**Comment by divan** [comment]
Ah, okay. I misunderstood `dispatch_sync()` behaviour then.

Do you have any ideas how to get task from DB as soon as it's inserted? Wait for update via RegisterCallback perhaps?

**Comment by hnvn** [comment]
Listen for Callback to trigger `loadTasks`, it might be a way to go in this case

**Comment by divan** [comment]
Yeah, but Callback will be executed quite often if many files are being downloaded, and that's a waste of CPU and I/O.

Do you think it's possible to wait until SQLite INSERT query is finished, and return from `enqueue()` only after that? To me it sounds logical – if DB write failed for any reason, I expect `enqueue()` to fail, not to return taskID.

**Comment by divan** [comment]
Okay, this is weird, but it seems like my problem was related to some issue of combination of IOS simulator and Flutter Hot Reload. Application Documents path is changing sometimes during Hot Reload, and that messes up everything.

Now I the code above returns correct tasks.

So my question now is, if `a new task is inserted into DB in a separate thread` is a guess or it's 100%. If latter, I still can not rely on this, and should implement logic differently.

**Comment by hnvn** [comment]
This article will give you information about Grand Center Dispatch: https://www.raywenderlich.com/5370-grand-central-dispatch-tutorial-for-swift-4-part-1-2

**Comment by divan** [comment]
Thanks, it seems to work properly so far. Closing issue for now.
